src/Observable.ts

```diff
@@ -35,19 +35,78 @@ export class Observable<T> extends BaseObservable<T> {
     }
 
     map<U>(action: (value: T) => U): Observable<U> {
-        throw new Error('not implemented')
+        return new Observable<U>(observer => {
+            this.subscribe(
+                value => {
+                    if (observer.closed) {
+                        return
+                    }
+
+                    let newValue
+                    try {
+                        newValue = action(value)
+                    } catch (e) {
+                        return observer.error(e)
```

Comment:
Jak w try użyje `observer.next(action(value))` a tutaj `observer.error(e)` to rozumiem że mogę sobie odpuscić `returna` :)